pkg/gateway/vald/handler/grpc/handler.go

```diff
@@ -593,19 +616,56 @@ func (s *server) MultiUpsert(ctx context.Context, vecs *payload.Object_Vectors)
 			span.End()
 		}
 	}()
-	metas := make([]string, 0, len(vecs.GetVectors()))
+
+	insertVecs := make([]*payload.Object_Vector, 0, len(vecs.GetVectors()))
+	updateVecs := make([]*payload.Object_Vector, 0, len(vecs.GetVectors()))
+
 	for _, vec := range vecs.GetVectors() {
-		metas = append(metas, vec.GetId())
-	}
-	uuids, err := s.metadata.GetUUIDs(ctx, metas...)
-	if err == nil {
-		for i := range uuids {
-			if len(uuids) > i && len(uuids[i]) != 0 {
-				return s.MultiUpdate(ctx, vecs)
+		exists, err := s.metadata.Exists(ctx, vec.GetId())
+		if err != nil {
```

Comment:
I think if upsert Exists returns error it's okay to add insertVecs

Reply:
@kpango I don't think so.
Because `s.metadata.Exists` does not return NotFound error here. NotFound error in `s.metadata.Exists` is treated as `err = nil` and returns `exists = false`.

This `err` may be a network error or something critical error in meta component.
Even if we continue to process, meta will return error in the following processes and the request will be failed.

Reply:
I know about it but this function is Multiple Upsert or Stream Upsert if we return error caused by only one Exists error the function is not useful, for upsert user don't want to care about data storeing even there object is exists or not which means we just skip or insert error vector.

Reply:
Revised in dcdccd03fa93124dec8e4da6883aabd46bcad844. @kpango